app/components/wallet-overview/wallet-overview.tsx

```diff
@@ -98,27 +85,39 @@ const WalletOverview: React.FC<Props> = ({ loading, setIsStablesatModalVisible }
       data?.me?.defaultAccount?.usdWallet?.balance ?? NaN,
     )
 
-    btcInDisplayCurrencyFormatted = moneyAmountToDisplayCurrencyString(btcWalletBalance)
-
-    usdInDisplayCurrencyFormatted = moneyAmountToDisplayCurrencyString(usdWalletBalance)
+    btcInDisplayCurrencyFormatted = `~ ${moneyAmountToDisplayCurrencyString(
+      btcWalletBalance,
+    )}`
 
     btcInUnderlyingCurrency = formatMoneyAmount({ moneyAmount: btcWalletBalance })
 
     if (displayCurrency !== WalletCurrency.Usd) {
       usdInUnderlyingCurrency = formatMoneyAmount({ moneyAmount: usdWalletBalance })
     }
+
+    usdInDisplayCurrencyFormatted = `${
+      usdInUnderlyingCurrency ? "~" : ""
```

Comment:
the logic of whether the ~ sign should be present or not should not be in the component imo. it should be in some function in the `useDisplayCurrency` hooks.

Reply:
IMO, I think the logic for adding the tilde sign (~) should be on the component level. We want the results of hooks to be pure and reusable in various components. Also, the computation for `usdInDisplayCurrencyFormatted = moneyAmountToDisplayCurrencyString(usdWalletBalance)` as well as the other two balances (`btcInDisplayCurrencyFormatted`, `usdInUnderlyingCurrency` and `btcInUnderlyingCurrency`) use functions from the `useDisplayCurrency` hook and they're done in the component level. Except we want to move these balance computations into the hook so that we get the values already formatted for just this screen.

Reply:
`formatMoneyAmount` takes in an argument called `isApproximate` that will add the `~`. I think it makes sense for the component to say if it is approximate, however all approximate numbers should be formatted the same way which is why it is part of the formatMoneyAmount fn.

Reply:
I wasn't aware of that functionality. Thanks.
I have refactored the code here https://github.com/GaloyMoney/galoy-mobile/pull/1920/commits/9d3b9fb730aad1c4b5919c75e30dd36e44f5e39a
cc @nicolasburtey

Reply:
fixed the duplicated code here https://github.com/GaloyMoney/galoy-mobile/pull/1920/commits/1694bd7bc72af327272893a20643b85bf72b9605